community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java

```diff
@@ -889,7 +939,11 @@ private void before( Object attachment, Callback cb )
             cb.started( attachment );
         }
 
-        if( interruptCounter.get() > 0 )
+        if ( terminationMark.get() )
```

Comment:
Maybe we can also use the interruptCounter with termination so that we can move this potentially costly check into the interruptCounter if-case? It would just feel better if we do not introduce unnecessary performance impacts.